src/Greenshot.Editor/Drawing/EmojiContainer.cs

```diff
@@ -0,0 +1,258 @@
+            if (_emojiPickerHost == null)
+            {
+                _emojiPicker = new Picker();
+                _emojiPicker.Picked += (_, args) =>
+                {
+                    _currentContainer.Emoji = args.Emoji;
+                    _currentContainer.Invalidate();
+                };
+
+                _emojiPickerHost = new ElementHost();
+                _emojiPickerHost.Dock = DockStyle.None;
+                _emojiPickerHost.Child = _emojiPicker;
+                _emojiPickerHost.Name = "EmojiPickerHost";
+
+                _parent.Controls.Add(_emojiPickerHost);
+            }
+        }
+
+        private void HideEmojiPicker()
+        {
+            _emojiPicker?.ShowPopup(false);
+        }
+
+        protected override void OnDeserialized(StreamingContext streamingContext)
+        {
+            base.OnDeserialized(streamingContext);
+            Init();
+        }
+
+        private void Init()
+        {
+            CreateDefaultAdorners();
+
+            // Create WPF control that will be used to render the emoji
+            _image = new System.Windows.Controls.Image();
+            global::Emoji.Wpf.Image.SetSource(_image, Emoji);
+
+            PropertyChanged += OnPropertyChanged;
+        }
+
+        private void OnPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName.Equals(nameof(Selected)))
+            {
+                if (!Selected)
+                {
+                    HideEmojiPicker();
+                }
+                else if (Status == EditStatus.IDLE && _justCreated)
+                {
+                    // Show picker just after creation
+                    ShowEmojiPicker();
+                    _justCreated = false;
+                }
+            }
+        }
+
+        public override void Draw(Graphics graphics, RenderMode rm)
+        {
+            graphics.SmoothingMode = SmoothingMode.HighQuality;
+            graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
+            graphics.CompositingQuality = CompositingQuality.HighQuality;
+            graphics.PixelOffsetMode = PixelOffsetMode.HighQuality;
+
+            var rect = GuiRectangle.GetGuiRectangle(Left, Top, Width, Height);
+
+            var iconSize = Math.Min(rect.Width, rect.Height);
+
+            if (iconSize > 0)
+            {
+                _image.Measure(new Size(iconSize, iconSize));
+                _image.Arrange(new Rect(0, 0, iconSize, iconSize));
+
+                var renderTargetBitmap = new RenderTargetBitmap(iconSize, iconSize, 96, 96, PixelFormats.Pbgra32);
+                renderTargetBitmap.Render(_image);
+
+                using var bitmap = BitmapFromSource(renderTargetBitmap);
+
+                var rotationAngle = GetRotationAngle();
+                if (rotationAngle != 0)
+                {
+                    graphics.DrawImage(RotateImage(bitmap, rotationAngle), Bounds);
```

Comment:
I found a memory leak here, you will need to dispose the RotateImage Bitmap after using it.

Reply:
Good catch!